src/shared/hooks/wallets.ts

```diff
@@ -270,3 +275,29 @@ export function useWalletChange() {
     isStaked,
   ])
 }
+
+export function useCachedWalletName() {
+  const [accountName, setAccountName] = useState<string | null>(null)
+  const walletAddress = useDappSelector(selectWalletAddress)
+
+  useEffect(() => {
+    const handleCachedNamesUpdate = () => {
+      if (!walletAddress) return
+
+      const cachedNames = localStorage.getItem(LOCAL_STORAGE_CACHED_NAMES)
+      if (!cachedNames) return
+
+      const parsedCachedNames: CachedNames = JSON.parse(cachedNames)
+      const { ens, uns } = parsedCachedNames[walletAddress]
+
+      if (ens) setAccountName(ens.name)
+      if (uns) setAccountName(uns.name)
```

Comment:
ENS names should override UNS names if user has both (same as we have it in the wallet)